File: src/NumPrimo.java
import java.util.Scanner;

public class NumPrimo {
        public static void main(String[] args) {
            Scanner scanner = new Scanner(System.in);
            System.out.print("Digite um número: ");
            int numero = scanner.nextInt();
            scanner.close();

            if (!ehPrimo(numero)) {
                System.out.println("O número " + numero + " NÃO é primo.");
            } else {
                System.out.println("O número " + numero + " é primo.");
            }
        }

        public static boolean ehPrimo(int num) {
            if (num < 2) return false;
            for (int i = 2; i <= Math.sqrt(num); i++) {
                if (num % i == 0) {
                    return false;
                }
            }
            return true;
        }


}

File: src/NumPrimo.js
const prompt = require('prompt-sync')({ sigint: true });

function ehPrimo(num) {
  if (num < 2) return false;
  for (let i = 2; i <= Math.sqrt(num); i++) {
    if (num % i === 0) {
      return false;
    }
  }
  return true;
}

function main() {
  const numero = parseInt(prompt("Digite um número: "));

  if (!ehPrimo(numero)) {
    console.log(`O número ${numero} NÃO é primo.`);
  } else {
    console.log(`O número ${numero} é primo.`);
  }
}

main();